cloudinit/net/network_state.py

```diff
@@ -69,7 +69,7 @@ def parse_net_config_data(net_config, skip_broken=True):
         # pass the whole net-config as-is
         config = net_config
 
-    if version and config:
+    if version and config is not None:
         nsi = NetworkStateInterpreter(version=version, config=config)
         nsi.parse_config(skip_broken=skip_broken)
         state = nsi.get_network_state()
```

Comment:
i don't quite get it, the result here is still that state is returned as `None`
where did the trace happen?

Reply:
When config was an empty list vs explicitly None

```
version: 1
config: []
```